.travis.yml

```diff
@@ -80,86 +65,59 @@ install:
   fi
 - export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/.local/lib
 - export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/.local/lib/pkgconfig
-- |
-  if [[ "$FEATURE" == "non-fatal-checks" ]]; then
-    cargo-outdated -V | grep $OUTDATED_VERS || cargo install cargo-outdated --vers $OUTDATED_VERS --force
-    cargo-deadlinks -V | grep $DEADLINKS_VERS || cargo install cargo-deadlinks --vers $DEADLINKS_VERS --force
-    fi
-- |
-  if [[ "$FEATURE" == "fmt" ]]; then
-    cargo-audit -V || cargo install cargo-audit --force
-    rustfmt -V | grep $RUSTFMT_VERS || cargo install rustfmt --vers $RUSTFMT_VERS --force
-    nvm install 8
-    nvm use 8
-    npm install -g cspell
-    cspell --version
-    npm install -g markdownlint-cli
-    markdownlint --version
-    fi
-- |
-  case "$FEATURE" in
-  clippy* )
-    cargo clippy --version | grep $CLIPPY_VERS || cargo install clippy --force --vers $CLIPPY_VERS
-  ;;
-  esac
-- cargo install --list
 
-script: |
-  case "$FEATURE" in
+jobs:
+  include:
+  # Formatting & other lints that do not require compilation
+  - env: FEATURE=fmt
+    install:
+    - cargo-audit -V || cargo install cargo-audit --force
+    - rustfmt -V | grep $RUSTFMT_VERS || cargo install rustfmt --vers $RUSTFMT_VERS --force
+    - nvm install 8 && nvm use 8
+    - npm install cspell
+    - ./node_modules/.bin/cspell --version
+    - npm install markdownlint-cli
+    - ./node_modules/.bin/markdownlint --version
+    script:
+    - cd exonum && cargo fmt -- --write-mode=diff && cd ..
+    - cd sandbox && cargo fmt -- --write-mode=diff && cd ..
+    - cd testkit && cargo fmt -- --write-mode=diff && cd ..
+    - cd services/time && cargo fmt -- --write-mode=diff && cd ../..
+    - ./node_modules/.bin/cspell sandbox/{src,examples,tests}/**/*.rs
+    - ./node_modules/.bin/cspell exonum/{src,benches,tests}/**/*.rs
+    - ./node_modules/.bin/cspell testkit/{src,examples,tests}/**/*.rs
+    - ./node_modules/.bin/cspell services/time/{src,examples,tests}/**/*.rs
+    - find . -not -path "./3rdparty/*" && "./node_modules/*" -name "*.md" | xargs ./node_modules/.bin/markdownlint --config .markdownlintrc
+
+  # Clippy linting
+  - &clippy-core
+    env: FEATURE=clippy-core
+    rust: nightly-2018-01-22
+    install:
+    - cargo clippy --version | grep $CLIPPY_VERS || cargo install clippy --force --vers $CLIPPY_VERS
+    script:
+    - cargo clippy --all
+
+  # Tests
+  - env: FEATURE=test-core
+    script:
+    - cargo test -- all
```

Comment:
Should be `--all`. Also, you still need to run examples (I suggest to run examples in separate commands to get the full picture of possible failures).

Reply:
Are we sure that we want to *run* examples in CI? This will prevent us from creating an example which runs a server.

`cargo tests` builds the examples, and I think that is as good as you can get without restricting what examples should do?

Reply:
No, it's not possible to run several binaries (example or usual ones) with a single command: unlike tests, they can not terminate, and they can use different CLI interface.